examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -26,30 +14,57 @@
 #error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_ must be defined
 #endif
 
-// #define _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
 
 #if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
 	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage1D
 	#else
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image1D
 	#endif
+
+	#define WG_X 256
+	#define WG_Y 1
+	#define WG_Z 1
 #elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
 	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage2D
 	#else
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image2D
 	#endif
+
+	#define WG_X 16
+	#define WG_Y 16
+	#define WG_Z 1
 #elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
 	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage3D
 	#else
 		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image3D
 	#endif
+
+	#define WG_X 8
+	#define WG_Y 8
+	#define WG_Z 4
 #else
 	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
 #endif
 
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_X_
+#define _NBL_GLSL_WORKGROUP_SIZE_X_ WG_X
+#endif
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_Y_
+#define _NBL_GLSL_WORKGROUP_SIZE_Y_ WG_Y
+#endif
+
```

Comment:
maybe common header for the workgroup size choice?

Reply:
hmm actually because you always intend to use with `CComputeBlit::getDefaultWorkgroupDims`, you can throw all this default workgroup size setting out, and just error if workgroup sizes are not defined and their multiple doesn't equal the required thing for the `lower_bound` (since the binary search is cooperative)